### ISSUE TITLE
Get created rules are wrong when requested on more than one model

### ISSUE THREAD

**Issue by kfirba** [jarektkaczyk/eloquence#16]
Hello.

I have some tests which call `getCreatedRules()` on a `User` model first and after few tests it's calling the `getCreatedRules()` on a `Profile` model and for some odd reason the rules that are being returned are the `User` rules. When I run the `Profile` tests alone, everything works just fine.

The weird thing is that tests shouldn't depend on each other but somehow one test affects the other?

a solution that I found for the time being:

```php
/**
* Get all the validation rules for this model.
*
* @return array
*/
public static function getCreateRules()
{
    if (!eval('\\'.get_called_class().'::$rulesMerged;')) {
        static::$rulesMerged = static::gatherRules();
    }

    return static::$rulesMerged;
}
```

another solution is to simply omit the `if` statement and always gather the rules but that can be expensive call. I wonder why this issue is happening. I was sure that the `static` keyboard will return the called class but it seems like it doesn't? the PHP version is 5.6.10

**Comment by kfirba** [comment]
I also have to mentioned that using the Validable trait cause many unexplained issues in my testing environment. Unfortunately I had to remove the validable trait in order to make my app functional again.

**Comment by jarektkaczyk** [comment]
@kfirba Does your `Profile` extend `User` model?

I think there's might be tiny bug concerning polymoprhysm, so I'll check this out.

**Comment by kfirba** [comment]
@jarektkaczyk My `Profile` extends the `Sofa\Eloquence\Model`. The validable trait exposed many breakages in my tests and I really tried to figure out why and what is going on for 3 days until I simply removed the trait and not everything works just as expected. If you need any detail, please let me know.

**Comment by jarektkaczyk** [comment]
OK, can you add some examples? I've used it in tests and had no issues, however testing depends on the tools.

**Comment by jarektkaczyk** [comment]
Also, take this into consideration: https://github.com/laravel/framework/issues/1181

This is just an example, however there were many issues with eloquent (events especially) during tests. Testing Eloquent models is tricky, that's a fact.

**Comment by kfirba** [comment]
@jarektkaczyk Sure. All of the issues happened with the new tests facilities that was presented to us in Laravel 5.1

Some issues:

1. The rules for Models after the first requested models were incorrect. If I gathered the rules for a User creation in test number 1 and then in test number 10 I wanted to gather the rules for Profile creation and I was returned with the User creation rules.
2. I'm not entirely sure that this is caused by the Validable trait, but when I used the Validable trait this happened: https://github.com/laravel/framework/issues/9733 . When I removed the Validable trait I could safely remove my suggested temporary solution and everything worked fine. I also tried to re-use the Validable trait without my suggested temporary solution (after ensuring that my tests passed without the temporary solution) and the issue was back there. 

These are the 2 side effects that drove me crazy as I didn't understand why my tests are failing.
